src/bin/pgcopydb/file_utils.h

```diff
@@ -54,6 +54,26 @@ bool write_file(char *data, long fileSize, const char *filePath);
 bool append_to_file(char *data, long fileSize, const char *filePath);
 bool read_file(const char *filePath, char **contents, long *fileSize);
 bool read_file_if_exists(const char *filePath, char **contents, long *fileSize);
+
+/* iterate over a file one line at a time */
+typedef bool (FileIterLinesFun)(void *context, char *line);
+
+bool file_iter_lines(const char *filename, size_t bufsize,
+					 void *context,
+					 FileIterLinesFun *callback);
```

Comment:
Why do we have a function declaration and definition if it is not used anywhere in the codebase?

Reply:
The idea is that we have other places where we read a file line-by-line that could benefit from this iterator and one-line-at-a-time in memory approach. Also to implement the full iterator protocol, even though the composition with the archive iterator needs to be “open-coded” at the moment, using the lower-level API directly.